curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentTtlNode.java

```diff
@@ -61,12 +61,15 @@ public class PersistentTtlNode implements Closeable {
     public static final int DEFAULT_TOUCH_SCHEDULE_FACTOR = 2;
     public static final boolean DEFAULT_USE_PARENT_CREATION = true;
 
+    private static final boolean SHUTDOWN_EXECUTOR = true;
+
     private final Logger log = LoggerFactory.getLogger(getClass());
     private final PersistentNode node;
     private final CuratorFramework client;
     private final long ttlMs;
     private final int touchScheduleFactor;
     private final ScheduledExecutorService executorService;
+    private final boolean shutdownExecutor; // If needed to shut down executorService when closing
```

Comment:
I saw `CloseableScheduledExecutorService`, and think it fit this use case pretty well.

Reply:
I think so

Reply:
> Probably I could extend that as part of the solution. I assume that is OK

That would be good!

Reply:
Included new commit.

As commented extended CloseableScheduledExecutorService  , and added some tests there.

Also fixed the recipe  and added some tests (maybe redundant with CloseableScheduledExecutorService tests